src/compas_timber/design/workflow.py

```diff
@@ -77,10 +161,19 @@ def __repr__(self):
             CategoryRule.__name__, self.joint_type.__name__, self.category_a, self.category_b, self.topos
         )
 
-    def comply(self, beams):
+    def comply(self, beams, max_distance=1e-6):
         try:
             beam_cats = set([b.attributes["category"] for b in beams])
-            return beam_cats == set([self.category_a, self.category_b])
+            comply = False
+            beams = list(beams)
+            if beam_cats == set([self.category_a, self.category_b]):
+                solver = ConnectionSolver()
+                if (
+                    self.joint_type.SUPPORTED_TOPOLOGY
+                    == solver.find_topology(beams[0], beams[1], max_distance=max_distance)[0]
```

Comment:
here you still use `find_topology` as an instance method, even though you made it static. I personally think it should be instance method again, but at the very least it should be used consistently

Reply:
I will revert to instance method. should all the `ConnectionSolver`all be instance methods?